doc/computing/parallelism.rst

```diff
@@ -194,6 +194,60 @@ These environment variables should be set before importing scikit-learn.
     Sets the seed of the global random generator when running the tests,
     for reproducibility.
 
+    Note that scikit-learn tests are expected to run deterministically with
+    explicit seeding of their own independent RNG instances instead of relying
+    on the numpy or Python standard library RNG singletons to make sure that
+    test results are independent of the test execution order. However some
+    tests might forget to use explicit seeding and this variable is a way to
+    control the intial state of the aforementioned singletons.
+
+:SKLEARN_TESTS_GLOBAL_RANDOM_SEED:
+
+    Controls the seeding of the random number generator used in tests that
+    rely on the `global_random_seed`` fixture.
+
+    All tests that use this fixture accept the contract that they should
+    deterministically pass for any seed value from 0 to 99 included.
+
+    If the SKLEARN_TESTS_GLOBAL_RANDOM_SEED environment variable is set to
+    "any" (which should be the case on nightly builds on the CI), the fixture
+    will choose an arbitrary seed in the above range (based on the BUILD_NUMBER
+    or the current day) and all fixtured tests will run for that specific seed.
+    The goal is to ensure that, over time, our CI will run all tests with
+    different seeds while keeping the test duration of a single run of the full
+    test suite limited. This will check that the tests assertions of tests
+    written to use this fixture are not dependent on a specific seed value.
```

Comment:
```suggestion
    test suite limited. This will check that the assertions of tests
    written to use this fixture are not dependent on a specific seed value.
```